storageClient/api.go

```diff
@@ -113,6 +114,27 @@ type GetMembershipKeyRequest struct {
 	GroupID uuid.UUID `json:"group_id"`
 }
 
+// ListGroupRecordsRequest wraps values needed for the request to list all records shared with a group
+type ListGroupRecordsRequest struct {
+	GroupID   uuid.UUID `json:"group_id"`
+	NextToken int64
+	Max       int
+}
+
+// ExecutorQueryResponse contains the high level information for the response sent to client by the executor.
+type ExecutorQueryResponse struct {
+	ResultList   []pdsClient.ListedRecord `json:"results"`
+	LastIndex    int                      `json:"last_index"`
+	TotalResults int64                    `json:"total_results"`
```

Comment:
If we have this that means on the server side we need to do a query to count the total number of records shared,otherwise it needs to be removed

Reply:
Do we want this behavior?

Reply:
Want yes, need, well that depends on how much work it is. I think it can be done similarly to the query to look up the group records themself, but instead of select * from its select count(*) from, if thats the case yes we need it, otherwise we can defer for now.